src/univariate/continuous/cosine.jl

```diff
@@ -65,12 +65,24 @@ function logpdf{T<:Real}(d::Cosine{T}, x::Real)
     insupport(d, x) ? log(pdf(d, x)) : -T(Inf)
 end
 
-function cdf(d::Cosine, x::Real)
+function cdf{T<:Real}(d::Cosine{T}, x::Real)
+    if x < d.μ - d.σ 
+        return zero(T)
+    end
+    if x > d.μ + d.σ 
+        return one(T)
+    end
     z = (x - d.μ) / d.σ
     (1 + z + sinpi(z) * invπ) / 2
 end
 
-function ccdf(d::Cosine, x::Real)
+function ccdf{T<:Real}(d::Cosine, x::Real)
+    if x < d.μ - d.σ 
+        return one(T)
+    end
+    if x > d.μ + d.σ 
+        return zero(T)
```

Comment:
My knowledge of Julia is very basic, so please excuse that I can't follow completely. Looking at say the [CDF of the uniform distribution](https://github.com/JuliaStats/Distributions.jl/blob/master/src/univariate/continuous/uniform.jl#L71), I fail to see the difference from the code I wrote. Could you elaborate?